src/messages/javascript.js

```diff
@@ -107,3 +107,15 @@ export const EVAL_STRING_ARG = {
     called only with function expressions as their first argument`),
   legacyCode: ['javascript', 'dangerous_global', 'eval'],
 };
+
+export const COMPLEX_PREFS_DEFAULTS = {
+  code: 'COMPLEX_PREFS_DEFAULTS',
+  message: _('Complex code should not appear in preference defaults files'),
+  description: _(singleLineString`Code in preference defaults files should
```

Comment:
I would change this description to something like: "Preference files are only allowed to call the pref() and user_pref() functions. Please move other code outside preference defaults file." (Feel free to remove the second second if it's TOO wordy)